### PR TITLE
failed to deploy PostgreSQL with operator on OCP 4.12

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Fixed: https://issues.redhat.com/browse/ACM-1950, https://github.com/stolostron/backlog/issues/26694
Reference: 
1. https://github.com/CrunchyData/postgres-operator-examples/issues/151
2. https://github.com/CrunchyData/postgres-operator/pull/3370

ErrorMessage:
```bash
time="2022-11-01T08:07:07Z" level=error msg="if kind is a CRD, it should be installed before calling Start" error="no matches for kind \"CronJob\" in version \"batch/v1beta1\"" file="sigs.k8s.io/controller-runtime@v0.8.3/pkg/log/deleg.go:144" func="log.(*DelegatingLogger).Error" kind=CronJob.batch version=5.0.3-0
panic: no matches for kind "CronJob" in version "batch/v1beta1"

goroutine 1 [running]:
main.assertNoError(...)
    github.com/crunchydata/postgres-operator/cmd/postgres-operator/main.go:39
main.main()
    github.com/crunchydata/postgres-operator/cmd/postgres-operator/main.go:84 +0x425
```

### CODE REVIEW

**Comment by yanmxa** [comment]
Postgres Operator issues:
```bash
The CustomResourceDefinition "postgresclusters.postgres-operator.crunchydata.com" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```
References: https://github.com/CrunchyData/postgres-operator-examples/issues/131

File: operator/config/samples/storage/deploy_postgres.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ fi
 
 # install postgres operator
 POSTGRES_OPERATOR=${POSTGRES_OPERATOR:-"pgo"}
-kubectl apply -k ${currentDir}/postgres-operator
+kubectl apply --server-side -k ${currentDir}/postgres-operator
```
Comment (clyang82) [inline operator/config/samples/storage/deploy_postgres.sh:15]: which `kubectl` version we are using?

Comment (yanmxa) [reply operator/config/samples/storage/deploy_postgres.sh:15]: The current `kubectl` is installed by this: https://github.com/stolostron/multicluster-global-hub/blob/35864cf33d1132fd0ece7e6eda8e7744ce0618b5/test/setup/e2e_dependencies.sh#L27
The client version is  `v1.18.0` and the server version is `v1.23.6`
```
$ kubectl version 
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6", GitCommit:"ad3338546da947756e8a88aa6822e9c11e7eac22", GitTreeState:"clean", BuildDate:"2022-05-19T19:39:28Z", GoVersion:"go1.17.9", Compiler:"gc", Platform:"linux/amd64"}
```

Comment (clyang82) [reply operator/config/samples/storage/deploy_postgres.sh:15]: so it seems `--server-side` is default value. we do not need to add it.

Comment (yanmxa) [reply operator/config/samples/storage/deploy_postgres.sh:15]: If we don't add the flag. The following message has occurred:
```bash
The CustomResourceDefinition "postgresclusters.postgres-operator.crunchydata.com" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```
ref: https://github.com/CrunchyData/postgres-operator-examples/issues/131

**Pull request merged**